src/Pact/Types/Runtime.hs

```diff
@@ -247,6 +249,11 @@ data EvalEnv e = EvalEnv {
 makeLenses ''EvalEnv
 
 
+toPactId :: TxId -> ChainId -> PactId
+toPactId txId (ChainId chainId) = (PactId . T.concat)
+                                  ["tx-id:",asString txId, ":",
```

Comment:
Went with Text and this structure, to make it more readable in case of errors, but open to suggestions.

Reply:
By transaction hash do you mean the hash generated here: https://github.com/kadena-io/pact/blob/master/src/Pact/Types/Command.hs#L141?

Reply:
yes, you need this function to run in `Eval e` so you can get to it via _eeHash: https://github.com/kadena-io/pact/blob/3abb051d9401ee02fc2a62ebcdff362d23edd58f/src/Pact/Types/Runtime.hs#L237

Reply:
Thus, `ask >>= \EvalEnv{..} -> hash (encode _eeTxId <> encode _eeHash <> encode _pmChainId (_pdPublicMeta _eePublicData))`

Reply:
See latest commits.